### DISCUSSION TITLE
How do you handle multi-layer header CSV file's series index ?

### DISCUSSION THREAD

**Discussion by coolbq** [johnkerl/miller#1467] [Q&A] [answered]
As titled, how Miller deal with this kind of files to extract series index and series data?

**Comment by coolbq** [comment]
> As titled, how Miller deal with this kind of files to extract series index and series data?

![微信图片_20240111090448](https://github.com/johnkerl/miller/assets/2296444/433a5b2c-8d45-43e8-8041-565c594078a5)

[sample.zip](https://github.com/johnkerl/miller/files/13896012/sample.zip)

**Comment by aborruso** [comment]
Can you add an example of the output you would like? Or the process you would like to run?

**Reply by coolbq** [reply]
I want to seperate the first 6 lines as series index, and the data in rows with no bigger than 6 as the series data. And finally to save them in two seperate files. Than I can visualize the data in tools like Tableau.

**Reply by aborruso** [reply, marked as answer]
I'm not sure I understand, but I'll try to give you an answer.

Running `<sample.tsv tail -n +6 >body.tsv`, you save the data. 
Running `<sample.tsv head -n 5 >header.tsv` you save the header.

The body will be in this way

![image](https://github.com/johnkerl/miller/assets/30607/a4d96531-22b3-455e-805a-671a7c943f02)

If it is useful for you, you can transpose the header, coupling Miller with [qsv](https://github.com/jqnatividad/qsv).

```
<header.tsv mlr --csv remove-empty-columns | qsv transpose >transposed_header.tsv
```

![image](https://github.com/johnkerl/miller/assets/30607/a80f2119-8f6c-4439-95a9-cca4b9bb5cbb)

**Reply by coolbq** [reply]
Thanks ! aborruso, your suggested method solved it. It's really a great idea to split before processing. It help alot.
